components/shared/script_layout/wrapper_traits.rs

```diff
@@ -285,8 +285,15 @@ pub trait ThreadSafeLayoutNode<'dom>: Clone + Copy + Debug + NodeInfo + PartialE
 
     fn node_text_content(self) -> Cow<'dom, str>;
 
+    /// If there's selection within this node, returns it. Otherwise, returns `None`.
+    fn selection(&self) -> Option<Range<usize>>;
+
+    /// If there's selection within this node in, returns it in ByteIndex. Otherwise, returns `None`.
+    /// This is used in the old layout
+    fn selection_byteindex(&self) -> Option<range::Range<ByteIndex>>;
+
     /// If the insertion point is within this node, returns it. Otherwise, returns `None`.
-    fn selection(&self) -> Option<Range<ByteIndex>>;
```

Comment:
There's a slight complication that the old layout uses `range::Range<ByteIndex>` instead of the use `std::ops::Range<usize>` that the underlying HTML elements uses.

Personally I feel like there's no reason to do the cast for usage in the new layout. But I'm not sure if we're allowed(?) to just change this.

Reply:
The updated selection method you wrote make more sense in CJK context. I'm happy to see the change but I suppose this need people who maintain layout to approve.

CC @Loirooriol

Reply:
@dklassic After reviewing, it seems possible to use existing ByteIndex to achieve what you want. Could you keep the existing trait method and see if it could do the work?